### PR TITLE
Meeting query implementation

### PR DESCRIPTION
**Pull request by proyectoramirez**
These changes implement the FindMeetingQuery class in weeks 5 project, which finds available time in the calendar of a given set of attendees for a meeting of a given amount of time. These changes pass all the pre-defined tests.

Demo: https://meeting-query-implementation-dot-josedramirez-step-2020.appspot.com

This branch may be merged to master in no particular order.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,80 @@
...
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> unavailableTimeRanges = getUnavailableTimes(events, request.getAttendees());
+    List<TimeRange> suggestedTimeRanges = findAvailableTimeRanges(unavailableTimeRanges, request.getDuration());
+
+    return suggestedTimeRanges;
+  }
+
+  private List<TimeRange> getUnavailableTimes(Collection<Event> events, Collection<String> attendees) {
```
Comment (Enriqe) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:53]: nit: please add Javadoc to all methods

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,80 @@
...
+      int lastIndex = mergedTimeRanges.size() - 1;
+      TimeRange lastTimeRange = mergedTimeRanges.get(lastIndex);
+      
+      if (lastTimeRange.contains(timeRange)) {
+        continue;
+      } else if (lastTimeRange.overlaps(timeRange)) {
+        mergedTimeRanges.set(lastIndex, TimeRange.fromStartEnd(lastTimeRange.start(), timeRange.end(), false));
+      } else {
+        mergedTimeRanges.add(timeRange);
+      }
```
Comment (Enriqe) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]:
```suggestion
      if (lastTimeRange.contains(timeRange)) {
        continue;
      } 
      
      if (lastTimeRange.overlaps(timeRange)) {
        mergedTimeRanges.set(lastIndex, TimeRange.fromStartEnd(lastTimeRange.start(), timeRange.end(), false));
        continue;
      } 
      
      mergedTimeRanges.add(timeRange);
```

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -14,10 +14,83 @@
...
+        continue;
+      } 
+      
+      mergedTimeRanges.add(timeRange);
+    }
+
+    return mergedTimeRanges;
+  }
+
+  private List<TimeRange> findAvailableTimeRanges(List<TimeRange> unavailableTimeRanges, long eventDuration) {
```
Comment (newmuis) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:129]: nit: it's good to include the unit of measurement in the name of `eventDuration` (e.g. `eventDurationMs` or `eventDurationMillis`)

**Review by newmuis — DISMISSED** [review]

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -14,10 +14,137 @@
...
+  /**
+   * Given a collection of time ranges, this will merge those that overlap.
+   * 
+   * @param timeRanges An unsorted collection of time ranges.
+   * 
+   * @return A list of merged time ranges, sorted by start time.
+   */
+  private List<TimeRange> mergeOverlappingTimeRanges(Collection<TimeRange> timeRanges) {
+    List<TimeRange> sortedTimeRanges = timeRanges
+        .stream()
```
Comment (Enriqe) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:88]: Nice 👍

**Review by Enriqe — APPROVED** [review]
💯

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 5]
```diff
@@ -14,10 +14,137 @@
...
+   * @param attendees A collection of people to consider when calculating
+   *    busy time.
+   * 
+   * @return A list of time ranges for which anyone in the list of people are busy,
+   *    sorted by start time.
+   */
+  private List<TimeRange> getUnavailableTimes(Collection<Event> events, Collection<String> attendees) {
+    List<TimeRange> eventTimes = events
+        .stream()
+        .filter(event -> !areAttendeesInEvent(event, attendees))
```
Comment (Enriqe) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:56]: 👍

**Pull request merged**